decidim-forms/app/queries/decidim/forms/questionnaire_participants.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Forms
+    # A class used to collect user answers for a questionnaire
+    class QuestionnaireParticipants < Rectify::Query
+      # Syntactic sugar to initialize the class and return the queried objects.
+      #
+      # questionnaire - a Questionnaire object
+      def self.for(questionnaire)
+        new(questionnaire).query
+      end
+
+      # Initializes the class.
+      #
+      # questionnaire = a Questionnaire object
+      def initialize(questionnaire)
+        @questionnaire = questionnaire
+      end
+
+      # Finds all participants (unique session_tokens).
+      def query
+        Answer.where(questionnaire: @questionnaire)
```

Comment:
is this correct?
It says it finds all participants but it returns the Answers in the Questionnaire..

Reply:
Hi @tramuntanal! Yes, I understand the naming is confusing, but the only way to retrieve the participant's info is through the answers, and we thought it would be too big a change to modify the answers table in the DB.